birdhouse/optional-components/secure-thredds/secure-access-magpie-permission.cfg

```diff
@@ -36,15 +36,18 @@ permissions:
     group: anonymous
     action: create
 
-  # preserve access for test-suite user
+  # following permissions provide access to 'secure' directory to any user that is member of 'thredds-secure' group
+  # this access will take priority over 'anonymous' group above that gets denied public access to 'secure' directory
+  # users that must be granted access to the data under 'secure' should be added to the group and will inherit access
+
   - service: thredds
     resource: /birdhouse/testdata/secure
     type: directory
     permission:
       name: browse
       access: allow
       scope: recursive
-    user: authtest
+    group: thredds-secure
```

Comment:
Sorry for nickpicing but `thredds-secure` is potentially a very generic name.  It would be much clearer that it is used for testing if the group is called `authtest-group`.

Reply:
This is why I believe it should in fact be the generic name and not be `authtest-group`. 
There is nothing related to `authtest` specifically in that config, it just so happens that `authtest` user uses it because it wants to validate the same permission access this group provides. 
I would rather have a real group `thredds-secure`, that we could actually use to setup real users, that is also validated against the same user membership. We test the use case 1:1.

Reply:
> I would rather have a real group `thredds-secure`, that we could actually use to setup real users

This is exactly what I want to avoid.  This is a generic repo, we should not be implementing security for all the org deploying this repo.  And I would not want "real user" be part of this group.  Otherwise the `authtest` user will start to have more permissions as a "real user" would, which will be bad.

We are replacing the `authtest` user with a group with the exact same intention here.

Reply:
What makes it not generic? 

If you enable *only* component `secure-thredds`, you receive a set of very specific permissions on Thredds service, and with it the corresponding group to give users membership *if you want to*. Using `authtest-group` there means you would have a random group named `authtest-group` when you are not even using `authtest` user for tests.

If instead you have the `authtest` user creation script, it is because you intend to test features, which includes testing the behaviour provided by `secure-thredds`. It makes perfect sense that the user used to test that security feature joins the relevant group created for that purpose. This is the same as any other "public" users joining the `anonymous` group to evaluate access to publicly accessible services.

Using the same group also ensures that we test everything the same way as the component defines it, regardless of who `authtest` actually is. I could call it `TheLegend27` for all we care, but it would still inherit the correct permissions to evaluate `secure-thredds` feature if he is member of `thredds-secure` group.

Reply:
I think we all agree this component is meant for testing.

So just like naming classes or variable names in a program, we want to have a descriptive name that reflect the actual role, which is for testing in the case of this `authtest-group`.  Matching the group name with the `authtest` user name will better relate each other.

On the production point of view, activating a component is a done once and forget thing.  For other orgs not heavily involved in the development like us, I think one year later, they might not remember what the group name `thredds-secure` was meant for so having a more descriptive name will help.

Or other orgs might actually want to use the fairly generic group name `thredds-secure` for something real, not for testing, thus I suggested `authtest-group` which will much less likely to cause name conflict in the future.

If we are both so uncomfortable with creating a group, then may I suggest another solution: perform a user update right after a user create so we ensure user `authtest` will always have the correct password and no need to create this group anymore.  This group was really just a work-around to ensure the user `authtest` have the correct password.  We do not actually need to have this group.

Reply:
In the same way as you describe to be precise about the role, `authtest-group` doesn't indicate immediately what access gets granted when you are a member of it. You still need to know the specific components enabled to know its application.
So I would suggest to go all out with `thredds-secure-authtest-group` since it is so specific for that situation.

I think the group is less trouble than the user create/update handling.  
It also allows to extend tests with more users later on if we need to.

Reply:
Sold for `thredds-secure-authtest-group` thanks.